### PR TITLE
Add share note functionality

### PR DESCRIPTION
**Pull request by pelumy**
## Issues covered
#1272

## Description
This PR adds a share functionality to the share note button option on the  confirmation dialogue in the `NoteOptionsButton` file. The following changes were made:

- A `showingShare` state variable was already present, so set it to `true` when the share note button is pressed.
- The share sheet was already present, so updated the code block to include sending the note link to the Apple Share sheet.
- Adds an optional completion handler to the `ActivityViewController` to perform actions after the share activity is finished.
- Sets `showingShare` to `false` in this completion handler to dismiss the share sheet after the share activity is finished.

## How to test
1. Build the app.
2. Go to the Feed tab.
3. Choose any note and select the 3 dots menu at the top right of the note, just beside the time.
4. Click the share note button option
5. Please confirm that Apple's share sheet comes up and you can share the note link anywhere.

## Screenshots/Video

https://github.com/user-attachments/assets/4eee13a6-2f6a-4eb6-9ffa-b5bbd0300db4

### CODE REVIEW

**Comment by pelumy** [comment]
I'm curious as to why this note link directs the user to an jump website instead of the nos app 🤔

File: Nos/Views/Components/ActivityView.swift [hunk 1]
```diff
@@ -5,11 +5,19 @@ struct ActivityViewController: UIViewControllerRepresentable {
 
     var activityItems: [Any]
     var applicationActivities: [UIActivity]?
+    var completion: (() -> Void)? // Completion handler called after the share activity is finished
```
Comment (joshuatbrown) [inline Nos/Views/Components/ActivityView.swift:8]: Could we make this a doc comment and simplify it slightly? Something like this (mostly copied from [Apple's docs](https://developer.apple.com/documentation/uikit/uiactivityviewcontroller/1622022-completionwithitemshandler)):

```suggestion
    /// The completion handler to execute after the share activity is finished.
    var completion: (() -> Void)? 
```

**Review by joshuatbrown — COMMENTED** [review]
This works perfectly! Weird but cool that we already had the `showingShare` property and sheet ready for you! Great work!

**Review by joshuatbrown — APPROVED** [review]
Thanks for the update! Perfect!

**Pull request merged**